src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php

```diff
@@ -399,14 +399,14 @@ private function describeValue($value, $omitTags, $showArguments)
             );
         }
 
-        if ($value instanceof Definition) {
-            return $this->getContainerDefinitionData($value, $omitTags, $showArguments);
-        }
-
         if ($value instanceof ArgumentInterface) {
             return $this->describeValue($value->getValues(), $omitTags, $showArguments);
         }
 
+        if ($value instanceof Definition) {
```

Comment:
but existing code might check for Definition. And in your PRs, you have ServiceLocatorArgument extending Definition but which should not be processed like a Definition.